runtime/shared_common/CacheMap.cpp

```diff
@@ -7455,3 +7709,250 @@ SH_CacheMap::isAddressInReleasedMetaDataBounds(J9VMThread* currentThread, UDATA
 	return rc;
 }
 
+/**
+ *  Get the unique ID of a pre-requisite cache at a lower layer
+ *
+ *	@param [in] currentThread The current JVM thread
+ *	@pamra [in] the cache directory
+ *	@param [in] ccToUse The current cache that depends on lower layer cache.
+ *	@param [in] startupForStats If the cache is started up for cache statistics
+ * 	@param [out] the Unique ID of a re-requiste cache.
+ * 	@param [out] the length of the Unique ID string.
+ *
+ *	@return 1 This cache depends on a lower layer cache. The unique ID of the re-requisite cache is found in or stored to this cache as metadata.
+ *			0 This cache does not depend on a low layer cache.
+ *			CM_CACHE_CORRUPT if cache is corrupted.
+ *			CM_READ_CACHE_FAILED if failed to get the existing pre-requisite cache ID from metadata.
+ *			CM_CACHE_STORE_PREREQ_ID_FAILED if failed to store the pre-requisite cache ID
+ *
+ *
+ */
+IDATA
+SH_CacheMap::getPrereqCache(J9VMThread* currentThread, const char* cacheDir, SH_CompositeCacheImpl* ccToUse, bool startupForStats, const char** prereqCacheID, UDATA* idLen)
+{
+	ShcItem* it = NULL;
+	IDATA result = 0;
+	SH_Manager* manager = NULL;
+	bool isReadOnly = ccToUse->isRunningReadOnly();
+	IDATA rc = 0;
+	PORT_ACCESS_FROM_PORT(_portlib);
+
+	Trc_SHR_Assert_True(ccToUse->hasWriteMutex(currentThread));
+
+	if ((UnitTest::CORRUPT_CACHE_TEST == UnitTest::unitTest)) {
```

Comment:
Please remove the redundant parentheses: will cause a warning with some compilers.

Reply:
Removed.